core/modules/comment/comment.module

```diff
@@ -1165,6 +1160,20 @@ function comment_form_node_type_form_alter(&$form, $form_state) {
       '#default_value' => $node_type->settings['comment_close_days'],
       '#field_suffix' => t('days'),
     );
+    $form['comment']['titles'] = array(
+      '#markup' => t('<label>Comment titles</label>'),
+    );
+    $form['comment']['comment_subject_field'] = array(
```

Comment:
I suggest switching the order of these two checkboxes, and hiding the `comment_subject_field` checkbox with states if the `comment_subject_hide` is checked, since it doesnt make much sense otherwise.

Reply:
I thought about this, but didn't do it because comments still need titles for internal use. HOWEVER, not what I think about it, they are very unlikely to need custom titles for internal use. So this idea does make sense.

Will implement it now.

Reply:
@docwilmot I disagree. There may be use-cases for wanting custom titles to be created, but not shown in comments themselves (perhaps they're used in admin views, etc.).

Reply:
> There may be use-cases for wanting custom titles to be created, but not shown in comments themselves (perhaps they're used in admin views, etc.).

In that case we generate the title as we do when they dont type one. There is no point to a UI where you ask a user to type a tile (in the normal usage, ie a comment thread) and it doesnt show. If you have a fancy UI that requires to show a title somewhere, then dont hide them. 

I quite frankly dont see a reasonable use case.

Reply:
I belive that we've resolved this with @BWPanda idea